frame/staking/src/pallet/impls.rs

```diff
@@ -731,19 +745,30 @@ impl<T: Config> Pallet<T> {
 			false
 		}
 	}
+
+	/// Register some amount of weight directly with the system pallet.
+	///
+	/// This is always mandatory weight.
+	fn register_weight(weight: Weight) {
```

Comment:
Maybe premature, but could we abstract to some helper function (kept somewhere in primitives?) and have it be generic over `T: frame_system::Config`? Then we could share it here and with election-provider-multiphase

Reply:
technically this abstraction exists: the public function in the system pallet which is accessible to everyone (`register_extra_weight_unchecked`). The fact that I preferred creating a helper function to avoid having to type `<frame_system::Pallet<T>>::...` twice is too opinionated for me to make a thing out of.